### ISSUE TITLE
Strange errors after upgrading to revapi 0.10.2 / 0.16.0

### ISSUE THREAD

**Issue by vmassol** [revapi/revapi#131]
We upgraded to 0.10.2 to fix the problems removed in this commit: https://github.com/xwiki/xwiki-platform/commit/a70c4652bfc76858b4289b5b2d43cfe579a49468#diff-f7d9f26837a9903e7c4befe7db15b03fL267

However we're now noticing the following new problems in our build (http://ci.xwiki.org/blue/organizations/jenkins/XWiki%2Fxwiki-platform/detail/master/676/pipeline/11 ):

```
[ERROR] Failed to execute goal org.revapi:revapi-maven-plugin:0.10.2:check (default-cli) on project xwiki-platform-notifications-filters-watch: The following API problems caused the build to fail:
[ERROR] java.method.finalMethodAddedToNonFinalClass: method java.lang.Class<?> java.lang.Object::getClass() @ org.xwiki.rendering.listener.MetaData: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::notify() @ org.xwiki.rendering.listener.MetaData: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::notifyAll() @ org.xwiki.rendering.listener.MetaData: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait() throws java.lang.InterruptedException @ org.xwiki.rendering.listener.MetaData: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait(long) throws java.lang.InterruptedException @ org.xwiki.rendering.listener.MetaData: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait(long, int) throws java.lang.InterruptedException @ org.xwiki.rendering.listener.MetaData: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method java.lang.Class<?> java.lang.Object::getClass() @ org.xwiki.rendering.listener.reference.ResourceReference: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::notify() @ org.xwiki.rendering.listener.reference.ResourceReference: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::notifyAll() @ org.xwiki.rendering.listener.reference.ResourceReference: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait() throws java.lang.InterruptedException @ org.xwiki.rendering.listener.reference.ResourceReference: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait(long) throws java.lang.InterruptedException @ org.xwiki.rendering.listener.reference.ResourceReference: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait(long, int) throws java.lang.InterruptedException @ org.xwiki.rendering.listener.reference.ResourceReference: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Throwable::addSuppressed(java.lang.Throwable) @ org.xwiki.rendering.parser.ParseException: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method java.lang.Class<?> java.lang.Object::getClass() @ org.xwiki.rendering.parser.ParseException: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method java.lang.Throwable[] java.lang.Throwable::getSuppressed() @ org.xwiki.rendering.parser.ParseException: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::notify() @ org.xwiki.rendering.parser.ParseException: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::notifyAll() @ org.xwiki.rendering.parser.ParseException: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait() throws java.lang.InterruptedException @ org.xwiki.rendering.parser.ParseException: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait(long) throws java.lang.InterruptedException @ org.xwiki.rendering.parser.ParseException: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait(long, int) throws java.lang.InterruptedException @ org.xwiki.rendering.parser.ParseException: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method java.lang.Class<?> java.lang.Object::getClass() @ org.xwiki.rendering.syntax.Syntax: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::notify() @ org.xwiki.rendering.syntax.Syntax: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::notifyAll() @ org.xwiki.rendering.syntax.Syntax: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait() throws java.lang.InterruptedException @ org.xwiki.rendering.syntax.Syntax: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait(long) throws java.lang.InterruptedException @ org.xwiki.rendering.syntax.Syntax: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait(long, int) throws java.lang.InterruptedException @ org.xwiki.rendering.syntax.Syntax: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method java.lang.Class<?> java.lang.Object::getClass() @ org.xwiki.rendering.syntax.SyntaxType: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::notify() @ org.xwiki.rendering.syntax.SyntaxType: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::notifyAll() @ org.xwiki.rendering.syntax.SyntaxType: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait() throws java.lang.InterruptedException @ org.xwiki.rendering.syntax.SyntaxType: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait(long) throws java.lang.InterruptedException @ org.xwiki.rendering.syntax.SyntaxType: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] java.method.finalMethodAddedToNonFinalClass: method void java.lang.Object::wait(long, int) throws java.lang.InterruptedException @ org.xwiki.rendering.syntax.SyntaxType: A final method has been added to an inheritable class. [<system classpath>]
[ERROR] 
[ERROR] If you're using the semver-ignore extension, update your module's version to one compatible with the current changes (e.g. mvn package revapi:update-versions). If you want to explicitly ignore this change and provide a justification for it, add the following JSON snippet to your Revapi configuration under "revapi.ignore" path:
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method java.lang.Class<?> java.lang.Object::getClass() @ org.xwiki.rendering.listener.MetaData",
[ERROR]   "package": "org.xwiki.rendering.listener",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.listener.MetaData",
[ERROR]   "classSimpleName": "MetaData",
[ERROR]   "methodName": "getClass",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::notify() @ org.xwiki.rendering.listener.MetaData",
[ERROR]   "package": "org.xwiki.rendering.listener",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.listener.MetaData",
[ERROR]   "classSimpleName": "MetaData",
[ERROR]   "methodName": "notify",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::notifyAll() @ org.xwiki.rendering.listener.MetaData",
[ERROR]   "package": "org.xwiki.rendering.listener",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.listener.MetaData",
[ERROR]   "classSimpleName": "MetaData",
[ERROR]   "methodName": "notifyAll",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait() throws java.lang.InterruptedException @ org.xwiki.rendering.listener.MetaData",
[ERROR]   "package": "org.xwiki.rendering.listener",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.listener.MetaData",
[ERROR]   "classSimpleName": "MetaData",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait(long) throws java.lang.InterruptedException @ org.xwiki.rendering.listener.MetaData",
[ERROR]   "package": "org.xwiki.rendering.listener",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.listener.MetaData",
[ERROR]   "classSimpleName": "MetaData",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait(long, int) throws java.lang.InterruptedException @ org.xwiki.rendering.listener.MetaData",
[ERROR]   "package": "org.xwiki.rendering.listener",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.listener.MetaData",
[ERROR]   "classSimpleName": "MetaData",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method java.lang.Class<?> java.lang.Object::getClass() @ org.xwiki.rendering.listener.reference.ResourceReference",
[ERROR]   "package": "org.xwiki.rendering.listener.reference",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.listener.reference.ResourceReference",
[ERROR]   "classSimpleName": "ResourceReference",
[ERROR]   "methodName": "getClass",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::notify() @ org.xwiki.rendering.listener.reference.ResourceReference",
[ERROR]   "package": "org.xwiki.rendering.listener.reference",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.listener.reference.ResourceReference",
[ERROR]   "classSimpleName": "ResourceReference",
[ERROR]   "methodName": "notify",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::notifyAll() @ org.xwiki.rendering.listener.reference.ResourceReference",
[ERROR]   "package": "org.xwiki.rendering.listener.reference",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.listener.reference.ResourceReference",
[ERROR]   "classSimpleName": "ResourceReference",
[ERROR]   "methodName": "notifyAll",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait() throws java.lang.InterruptedException @ org.xwiki.rendering.listener.reference.ResourceReference",
[ERROR]   "package": "org.xwiki.rendering.listener.reference",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.listener.reference.ResourceReference",
[ERROR]   "classSimpleName": "ResourceReference",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait(long) throws java.lang.InterruptedException @ org.xwiki.rendering.listener.reference.ResourceReference",
[ERROR]   "package": "org.xwiki.rendering.listener.reference",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.listener.reference.ResourceReference",
[ERROR]   "classSimpleName": "ResourceReference",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait(long, int) throws java.lang.InterruptedException @ org.xwiki.rendering.listener.reference.ResourceReference",
[ERROR]   "package": "org.xwiki.rendering.listener.reference",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.listener.reference.ResourceReference",
[ERROR]   "classSimpleName": "ResourceReference",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Throwable::addSuppressed(java.lang.Throwable) @ org.xwiki.rendering.parser.ParseException",
[ERROR]   "package": "org.xwiki.rendering.parser",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.parser.ParseException",
[ERROR]   "classSimpleName": "ParseException",
[ERROR]   "methodName": "addSuppressed",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method java.lang.Class<?> java.lang.Object::getClass() @ org.xwiki.rendering.parser.ParseException",
[ERROR]   "package": "org.xwiki.rendering.parser",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.parser.ParseException",
[ERROR]   "classSimpleName": "ParseException",
[ERROR]   "methodName": "getClass",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method java.lang.Throwable[] java.lang.Throwable::getSuppressed() @ org.xwiki.rendering.parser.ParseException",
[ERROR]   "package": "org.xwiki.rendering.parser",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.parser.ParseException",
[ERROR]   "classSimpleName": "ParseException",
[ERROR]   "methodName": "getSuppressed",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::notify() @ org.xwiki.rendering.parser.ParseException",
[ERROR]   "package": "org.xwiki.rendering.parser",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.parser.ParseException",
[ERROR]   "classSimpleName": "ParseException",
[ERROR]   "methodName": "notify",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::notifyAll() @ org.xwiki.rendering.parser.ParseException",
[ERROR]   "package": "org.xwiki.rendering.parser",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.parser.ParseException",
[ERROR]   "classSimpleName": "ParseException",
[ERROR]   "methodName": "notifyAll",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait() throws java.lang.InterruptedException @ org.xwiki.rendering.parser.ParseException",
[ERROR]   "package": "org.xwiki.rendering.parser",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.parser.ParseException",
[ERROR]   "classSimpleName": "ParseException",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait(long) throws java.lang.InterruptedException @ org.xwiki.rendering.parser.ParseException",
[ERROR]   "package": "org.xwiki.rendering.parser",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.parser.ParseException",
[ERROR]   "classSimpleName": "ParseException",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait(long, int) throws java.lang.InterruptedException @ org.xwiki.rendering.parser.ParseException",
[ERROR]   "package": "org.xwiki.rendering.parser",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.parser.ParseException",
[ERROR]   "classSimpleName": "ParseException",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method java.lang.Class<?> java.lang.Object::getClass() @ org.xwiki.rendering.syntax.Syntax",
[ERROR]   "package": "org.xwiki.rendering.syntax",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.syntax.Syntax",
[ERROR]   "classSimpleName": "Syntax",
[ERROR]   "methodName": "getClass",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::notify() @ org.xwiki.rendering.syntax.Syntax",
[ERROR]   "package": "org.xwiki.rendering.syntax",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.syntax.Syntax",
[ERROR]   "classSimpleName": "Syntax",
[ERROR]   "methodName": "notify",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::notifyAll() @ org.xwiki.rendering.syntax.Syntax",
[ERROR]   "package": "org.xwiki.rendering.syntax",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.syntax.Syntax",
[ERROR]   "classSimpleName": "Syntax",
[ERROR]   "methodName": "notifyAll",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait() throws java.lang.InterruptedException @ org.xwiki.rendering.syntax.Syntax",
[ERROR]   "package": "org.xwiki.rendering.syntax",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.syntax.Syntax",
[ERROR]   "classSimpleName": "Syntax",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait(long) throws java.lang.InterruptedException @ org.xwiki.rendering.syntax.Syntax",
[ERROR]   "package": "org.xwiki.rendering.syntax",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.syntax.Syntax",
[ERROR]   "classSimpleName": "Syntax",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait(long, int) throws java.lang.InterruptedException @ org.xwiki.rendering.syntax.Syntax",
[ERROR]   "package": "org.xwiki.rendering.syntax",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.syntax.Syntax",
[ERROR]   "classSimpleName": "Syntax",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method java.lang.Class<?> java.lang.Object::getClass() @ org.xwiki.rendering.syntax.SyntaxType",
[ERROR]   "package": "org.xwiki.rendering.syntax",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.syntax.SyntaxType",
[ERROR]   "classSimpleName": "SyntaxType",
[ERROR]   "methodName": "getClass",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::notify() @ org.xwiki.rendering.syntax.SyntaxType",
[ERROR]   "package": "org.xwiki.rendering.syntax",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.syntax.SyntaxType",
[ERROR]   "classSimpleName": "SyntaxType",
[ERROR]   "methodName": "notify",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::notifyAll() @ org.xwiki.rendering.syntax.SyntaxType",
[ERROR]   "package": "org.xwiki.rendering.syntax",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.syntax.SyntaxType",
[ERROR]   "classSimpleName": "SyntaxType",
[ERROR]   "methodName": "notifyAll",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait() throws java.lang.InterruptedException @ org.xwiki.rendering.syntax.SyntaxType",
[ERROR]   "package": "org.xwiki.rendering.syntax",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.syntax.SyntaxType",
[ERROR]   "classSimpleName": "SyntaxType",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait(long) throws java.lang.InterruptedException @ org.xwiki.rendering.syntax.SyntaxType",
[ERROR]   "package": "org.xwiki.rendering.syntax",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.syntax.SyntaxType",
[ERROR]   "classSimpleName": "SyntaxType",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] },
[ERROR] {
[ERROR]   "code": "java.method.finalMethodAddedToNonFinalClass",
[ERROR]   "new": "method void java.lang.Object::wait(long, int) throws java.lang.InterruptedException @ org.xwiki.rendering.syntax.SyntaxType",
[ERROR]   "package": "org.xwiki.rendering.syntax",
[ERROR]   "classQualifiedName": "org.xwiki.rendering.syntax.SyntaxType",
[ERROR]   "classSimpleName": "SyntaxType",
[ERROR]   "methodName": "wait",
[ERROR]   "elementKind": "method",
[ERROR]   "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
[ERROR] }
```

To reproduce, clone xwiki-platform and cd to `xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-watch` and run `mvn revapi:check`.

Any idea?

Thanks a lot Lukas :)

**Comment by metlos** [comment]
Strangely, I can't reproduce that...

**Comment by metlos** [comment]
I'm on xwiki-platform @ 85aa29198a04057cf9a963019ca18652eb3c6ee4 and cd-ing into the directory and running mvn revapi:check gives me a success...

**Comment by vmassol** [comment]
wow, you should be sleeping at this hour ;)

Thanks for looking into this.

You cannot reproduce probaby because I've added an ignore at https://github.com/xwiki/xwiki-platform/commit/f12a9f69e2fa0621bdc93952ea0ac1fc43ff2a3d :)

**Comment by vmassol** [comment]
> I'm on xwiki-platform @ 85aa29198a04057cf9a963019ca18652eb3c6ee4

Yep that's after after I added the ignore. You should remove it to reproduce the problem.

Thanks!

**Comment by metlos** [comment]
Thanks for letting me know.. I assume I woke you up, so good night :) I will have a look at this next week.. the weekend is full of birthday parties :)

**Comment by vmassol** [comment]
enjoy your weekend and parties :)

**Comment by metlos** [comment]
should be fixed in `revapi-java-0.17.1`.

**Comment by vmassol** [comment]
Awesome, thanks, will upgrade/test today.
